dag_in_context/src/lib.rs

```diff
@@ -113,28 +113,56 @@ pub(crate) fn print_with_intermediate_vars(termdag: &TermDag, term: Term) -> Str
 }
 
 // Returns a formatted string of (union call body) for each pair
-#[allow(dead_code)]
 fn print_function_inlining_pairs(
     function_inlining_pairs: Vec<function_inlining::CallBody>,
     printed: &mut String,
     tree_state: &mut TreeToEgglog,
     term_cache: &mut HashMap<Term, String>,
 ) -> String {
-    // Get unions
+    // Get unions and mark each call as inlined for extraction purposes
+    let inlined_call = "(relation InlinedCall (String Expr))";
     let unions = function_inlining_pairs
         .iter()
         .map(|cb| {
-            let call_term = cb.call.to_egglog_internal(tree_state);
-            let body_term = cb.body.to_egglog_internal(tree_state);
-            format!(
-                "(union {} {})",
-                print_with_intermediate_helper(&tree_state.termdag, call_term, term_cache, printed),
-                print_with_intermediate_helper(&tree_state.termdag, body_term, term_cache, printed)
-            )
+            if let Expr::Call(callee, _) = cb.call.as_ref() {
+                let call_term = cb.call.to_egglog_internal(tree_state);
+                let call_args = cb.call.children_exprs()[0].to_egglog_internal(tree_state);
+                let intermed_call_args = print_with_intermediate_helper(
+                    &tree_state.termdag,
+                    call_args.clone(),
+                    term_cache,
+                    printed,
+                );
+                let body_term = cb.body.to_egglog_internal(tree_state);
+                format!(
+                    // (union call body) (InlinedCall "callee" args) (subsume (Call "callee" args))
+                    // We need to subsume, otherwise the Call in the original program could get
+                    // substituted into another context during optimization and no longer match InlinedCall.
+                    "(union {} {})\n(InlinedCall \"{}\" {})\n(subsume (Call \"{}\" {}))\n",
+                    print_with_intermediate_helper(
```

Comment:
This can also be (Call {} {}), simplifying this code.
I suggest binding two variables:
`call_with_args` and `inlined`, and use them here.